common/aws/s3/fragment.go

```diff
@@ -82,6 +83,23 @@ func getFragmentKeys(fileKey string, fragmentCount int) ([]string, error) {
 	return keys, nil
 }
 
+// AllFragmentsExist returns true if all keys are fragment keys.
+// It checks if all fragment keys starting with 0th fragment and ending with nth fragment (marked by "f" postfix) exist.
+// Warning: this function sorts the input slice in place and therefore mutates the ordering of the input slice.
+func SortAndCheckAllFragmentsExist(keys []string) bool {
+	sort.Strings(keys)
+	for i, key := range keys {
+		if !strings.HasSuffix(key, "-"+strconv.Itoa(i)) {
+			if strings.HasSuffix(key, "-"+strconv.Itoa(i)+"f") {
+				return i == len(keys)-1
+			}
+			return false
+		}
+	}
+
+	return false
+}
```

Comment:
Do you intend for this to fail if there is a missing key in the middle? E.g. `abc-0`, `abc-1`, `--missing--`, `abc-3f`. The function `recombineFragments()` in `fragment.go` performs this sort of check, albeit on the `s3.Fragment` type, not on strings.